server/src/test/java/io/crate/analyze/expressions/ExpressionAnalyzerTest.java

```diff
@@ -238,6 +238,12 @@ public void testNonDeterministicFunctionsAlwaysNew() throws Exception {
         assertThat(fn1).isNotEqualTo(fn3);
     }
 
+    @Test
+    public void test_non_deterministic_function_is_not_normalized() {
+        Symbol symbol = executor.asSymbol("1 + random()");
+        assertThat(symbol).isFunction("add");
+    }
```

Comment:
This tests also passes on master.

I think we should instead test one of insert/update analyzers behavior change (and test should fail on master and pass here).

On master:
We always normalize copy/delete/insert/update as we pass `function -> true`.

On this PR: 
We normalize only non-deterministic functions (or functions with non-deterministic args). Can we check this?

Reply:
I updated comment, sorry for the confusion